src/script/sigcache.cpp

```diff
@@ -57,58 +43,51 @@ class CSignatureCache
     }
 
     bool
-    Get(const uint256& entry)
+    Get(const uint256& entry, const bool erase)
     {
         boost::shared_lock<boost::shared_mutex> lock(cs_sigcache);
-        return setValid.count(entry);
+        return setValid.contains(entry, erase);
     }
 
-    void Erase(const uint256& entry)
+    void Set(uint256& entry)
     {
         boost::unique_lock<boost::shared_mutex> lock(cs_sigcache);
-        setValid.erase(entry);
+        setValid.insert(entry);
     }
```

Comment:
To finish addressing @daira's `std::memory_order_relaxed` concern: `CSignatureCache::Get` and `CSignatureCache::Set` are the only locations where `CuckooCache` is accessed, and they use a `boost::unique_lock` to ensure that writes and reads are properly synchronized.